splunk/packages/edit-table/src/components/KVStoreUploader.jsx

```diff
@@ -40,17 +41,20 @@ export default function KVStoreUploader({
     };
     const handleFileChange = (fileContent) => setUploadedCSVContent(fileContent);
     const handleUploadCSV = async () => {
-        if (tableMetadata == null) {
-            return;
-        }
+        if (tableMetadata == null) return;
 
-        setUploading(true);
-        const { dataFields, totalItems } = tableMetadata;
         const emptyErrorMsg = 'No data to upload.';
         const emptyCSVErrorMsg = 'CSV file is empty.';
         const dataFieldErrorMsg = 'CSV file have no enough data fields.';
+        const backupErrorMsg = 'Error creating backup for KV Store. Please try again later.';
         const deleteErrorMsg = 'Error deleting all entries. Please try again.';
         const uploadErrorMsg = 'Error uploading csv. Please try again.';
+        const restoreErrorMsg = 'Error restoring KV Store. If data is lost, please contact admin.';
+
+        const { dataFields, totalItems } = tableMetadata;
+        let isBackupCreated = false;
+        let isUploadSuccess = false;
+        setUploading(true);
```

Comment:
you can simplify the whole code if you have a `doKvStoreChanges` function like this:
```js
try {
    await doKvStoreChanges(async () => {
        // doKvStoreChanges does first a backup of the kv store before calling these functions
        await deleteAllKVEntries(collectionName, deleteErrorMsg, splunkApp);
        await batchInsertKVEntries(
            collectionName,
            formattedJsonData,
            uploadErrorMsg,
            splunkApp
        );
    });
    // the callback finished successfully, show a success message
    setInfoMessage({
        visible: true,
        type: 'success',
        message: `CSV file successfully uploaded. Removed ${totalItems} items and added ${formattedJsonData.length} items.`,
    });
} catch (error) {
    // one of the kv store changes failed. doKvStoreChanges automatically restored the kv store backup and threw an error
    console.error(error);
    setInfoMessage({
        visible: true,
        type: 'error',
        message: error.message,
    });
}
```